src/utils/transactionUtils/transactionUtils.ts

```diff
@@ -326,10 +320,84 @@ export const FieldToInteractionMap: {
 } = {
   name: ANT_INTERACTION_TYPES.SET_NAME,
   ticker: ANT_INTERACTION_TYPES.SET_TICKER,
+  targetID: ANT_INTERACTION_TYPES.SET_TARGET_ID,
   // TODO: add other interactions
 };
 
 export function getInteractionTypeFromField(field: string) {
   // TODO: add contract specification and more interaction fields
   return FieldToInteractionMap[field];
 }
+
+export function mapTransactionDataKeyToPayload(
+  contractType: ContractType,
+  interactionType: AntInteraction | RegistryInteraction,
+  data: string[],
+) {
+  const payload: any = {};
+  try {
+    if (isRegistryInteraction(interactionType)) {
+      if (
+        !isInteractionCompatible({
+          contractType,
+          interactionType,
+          functionName:
+            TRANSACTION_DATA_KEYS[CONTRACT_TYPES.REGISTRY][interactionType]
+              .functionName,
+        })
+      ) {
+        throw new Error(`Interaction is not compatible`);
+      }
+      payload.functionName =
+        TRANSACTION_DATA_KEYS[CONTRACT_TYPES.REGISTRY][
+          interactionType
+        ].functionName;
+      TRANSACTION_DATA_KEYS[CONTRACT_TYPES.REGISTRY][
+        interactionType
+      ].keys.forEach((key, index) => {
+        if (!data[index]) {
+          // if missing transaction data from the url, throw an error
+          throw new Error(
+            `Missing key (${key}) from transaction data in the url. This may be due to the order of the data, the current order is [${data}], the correct order is [${
+              TRANSACTION_DATA_KEYS[CONTRACT_TYPES.REGISTRY][interactionType]
+                .keys
+            }]`,
+          );
+        }
+        payload[key] = data[index];
+      });
+    }
+    if (isAntInteraction(interactionType)) {
+      if (
+        !isInteractionCompatible({
+          contractType,
+          interactionType,
+          functionName:
+            TRANSACTION_DATA_KEYS[CONTRACT_TYPES.ANT][interactionType]
+              .functionName,
+        })
+      ) {
+        throw new Error(`Interaction is not compatible`);
+      }
+      payload.functionName =
+        TRANSACTION_DATA_KEYS[CONTRACT_TYPES.ANT][interactionType].functionName;
+      TRANSACTION_DATA_KEYS[CONTRACT_TYPES.ANT][interactionType].keys.forEach(
+        (key, index) => {
+          if (!data[index]) {
+            // if missing transaction data from the url, throw an error
+            throw new Error(
+              `Missing key (${key}) from transaction data in the url. This may be due to the order of the data, the current order is [${data}], the correct order is [${
+                TRANSACTION_DATA_KEYS[CONTRACT_TYPES.ANT][interactionType].keys
+              }]`,
+            );
+          }
+          payload[key] = data[index];
+        },
+      );
+    }
+  } catch (error) {
+    console.log(error);
+  } finally {
+    return payload; // eslint-disable-line
```

Comment:
why is eslint disabled? and what happens if this is an empty object?

Reply:
odd, not needed i guess.